src/Symfony/Component/Form/Util/StringUtil.php

```diff
@@ -39,4 +39,19 @@ public static function trim($string)
 
         return trim($string);
     }
+
+    /**
+     * Converts a fully-qualified class name to a block prefix.
+     *
+     * @param string $fqcn The fully-qualified class name
+     *
+     * @return string|null The block prefix or null if not a valid FQCN
+     */
+    public static function fqcnToBlockPrefix($fqcn)
+    {
+        // Non-greedy ("+?") to match "type" suffix, if present
+        if (preg_match('~([^\\\\]+?)(type)?$~i', $fqcn, $matches)) {
+            return strtolower(preg_replace('~([a-z])([A-Z])~', '$1_$2', $matches[1]));
```

Comment:
this may not handle class names containing a number properly. Look at the logic we have in the DI component (in ``Container::underscore``) to do the same kind of logic to generate DI aliases by default